webapp/src/features/Dashboard/ReportCard.js

```diff
@@ -4,100 +4,109 @@ import Cards from '../../widgets/Cards'
+			totalValue: 4.2,
+			from: 10.5,
+			successValue: 60.1,
+			upgrade: true,
+			data: [],
 		},
 		{
-			title: 'Lifetime Value',
-			graphValue: '$149',
-			range: '$555',
-			successValue: '27.01%',
+			totalValue: 149,
+			from: 555,
+			successValue: 27.01,
+			upgrade: true,
+			data: [],
 		},
+		{ totalValue: 200, from: 100, successValue: 200, upgrade: true, data: [] },
 		{
-			title: 'Average Revenue Per User',
-			graphValue: '$200',
-			range: '$100',
-			successValue: '200%',
-		},
-		{
-			title: 'Total MRR',
-			graphValue: '$50.5K',
-			range: '$50.4K',
-			successValue: '00.2%',
-		},
-		{
-			title: 'Active User Rate',
-			graphValue: '89.12%',
-			range: '78.50%',
-			successValue: '13.17%',
-		},
-		{
-			title: 'Growth Rate',
-			graphValue: '25.5%',
-			range: '24%',
-			successValue: '4.17%',
+			totalValue: 89.12,
+			from: 78.5,
+			successValue: 13.17,
+			upgrade: true,
+			data: [],
 		},
+		{ totalValue: 25.5, from: 24, successValue: 4.17, upgrade: true, data: [] },
+	]
+
+	!isEmpty(allTotalRevenue) && reportCards.splice(4, 0, allTotalRevenue)
+
+	const title = [
+		'User Churn Rate',
+		'Lifetime Value',
+		'Average Revenue Per User',
+		'Total MRR',
+		'Active User Rate',
+		'Growth Rate',
 	]
 
 	return (
-		<Col lg={8} xl={9}>
-			<Row>
-				{mockData.map((entry, index) => (
-					<Col key={entry.title} md={6} lg={6} xl={4}>
-						<Cards key={entry.title} title={entry.title}>
-							<div className='d-flex align-items-center justify-content-between mb-20'>
-								<div className='flex-fill d-flex'>
-									<h4 className='mb-0 gotham f-24 font-weight-bold graph-value'>
-										{entry.graphValue}
-									</h4>
-									<span className='f-12 align-self-end gotham opacity-50'>
-										{`From ${entry.range}`}
-									</span>
-								</div>
-								<div className='d-flex align-items-center'>
-									<h5
-										className={classNames(
-											'text-success gotham f-18 font-weight-bold mb-0 mr-1',
-											{
-												'text-danger': entry.successValue === '27.01%',
-											}
-										)}
-									>
-										{entry.successValue}
-									</h5>
-									<div className='arrow-bubble position-relative rounded-circle overflow-hidden d-flex align-items-center justify-content-center'>
-										<img
-											src={
-												entry.successValue === '27.01%'
-													? Icons.arrowDownIcon
-													: Icons.arrowUpIcon
-											}
-											alt=''
-											height='16'
-										/>
-										<span
+		<BusyIndicator>
+			<Col lg={8} xl={9}>
+				<Row>
+					{reportCards.map((entry, index) => (
+						<Col key={entry.title} md={6} lg={6} xl={4}>
+							<Cards key={title[index]} title={title[index]}>
+								<div className='d-flex align-items-center justify-content-between mb-20'>
+									<div className='flex-fill d-flex'>
+										<h4 className='mb-0 gotham f-24 font-weight-bold graph-value'>
+											{index === 0 || index === 4 || index === 5
```

Comment:
add the TODOs here
we need to remove the use of hardcoded indexes

Reply:
updated